<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Statistical Analysis with the `pmartR` Package • pmartR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Statistical Analysis with the `pmartR` Package">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pmartR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.4.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Convenience_Functions.html">Convenience Functions</a>
    </li>
    <li>
      <a href="../articles/Data_Normalization.html">Data Normalization</a>
    </li>
    <li>
      <a href="../articles/Data_Object_Creation.html">Data Object Creation</a>
    </li>
    <li>
      <a href="../articles/Filter_Functionality.html">Filter Functions</a>
    </li>
    <li>
      <a href="../articles/Graphing_with_pmartR.html">Graphing with pmartR</a>
    </li>
    <li>
      <a href="../articles/RNAseq_Processing_Workflow.html">RNAseq Data Processing</a>
    </li>
    <li>
      <a href="../articles/Statistical_Analysis_with_pmartR.html">Statistical Analysis with the `pmartR` Package</a>
    </li>
    <li>
      <a href="../articles/Trelliscope_Vignette.html">Using Trelliscope Functions with pmartR</a>
    </li>
    <li>
      <a href="../articles/Typical_Processing_Workflow.html">Typical Processing Workflow</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/pmartR/pmartR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Statistical Analysis with the
<code>pmartR</code> Package</h1>
                        <h4 data-toc-skip class="author">Bryan Stanfill,
Kelly Stratton, Lisa Bramer</h4>
            
            <h4 data-toc-skip class="date">2024-10-16</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/pmartR/pmartR/blob/v2.4.6/vignettes/Statistical_Analysis_with_pmartR.Rmd" class="external-link"><code>vignettes/Statistical_Analysis_with_pmartR.Rmd</code></a></small>
      <div class="hidden name"><code>Statistical_Analysis_with_pmartR.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://pmartr.github.io/pmartR/" class="external-link">pmartR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/pmartR/pmartRdata" class="external-link">pmartRdata</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">pep_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/edata_transform.html">edata_transform</a></span><span class="op">(</span></span>
<span>  omicsData <span class="op">=</span> <span class="fu">pmartRdata</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/pmartRdata/man/pep_object.html" class="external-link">pep_object</a></span>,</span>
<span>  data_scale <span class="op">=</span> <span class="st">"log2"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette describes the statistical analysis capabilities for
<code>omicsData</code> objects other than <code>seqData</code>. For
details about the statistical methods for <code>seqData</code> objects,
see <code><a href="../reference/diffexp_seq.html">?diffexp_seq</a></code>.</p>
<p>The <code>pmartR</code> package includes functions to assess
qualitative and quantitative differences in abundance data using the
IMD-ANOVA method described in <span class="citation">Webb-Robertson et
al. (2010)</span>. Our IMD-ANOVA functionality can handle omics datasets
with various experimental designs, including up to two grouping factors
with or without additional covariate information. The data can be paired
or not.</p>
<p>The functionality of the <code>pmartR</code> package will be
illustrated with peptide data available from the <code>pmartRdata</code>
package called <code>pep_object</code>. In a nutshell, the
<code>pep_object</code> object is a list of length three of the class
<code>pepData</code>. The <code>e_data</code> element contains all of
the peptide abundance data, the <code>f_data</code> object contains
sample information, and the <code>e_meta</code> object contains meta
information related to the peptides in the dataset. See
<code><a href="https://rdrr.io/pkg/pmartRdata/man/pmartRdata.html" class="external-link">?pmartRdata</a></code> for more on this data set.</p>
</div>
<div class="section level2">
<h2 id="imd-anova-test-for-differential-abundance">IMD-ANOVA Test for Differential Abundance<a class="anchor" aria-label="anchor" href="#imd-anova-test-for-differential-abundance"></a>
</h2>
<p>The <code><a href="../reference/imd_anova.html">imd_anova()</a></code> function is used to implement the IMD
test, ANOVA test, or combined IMD-ANOVA test. Using the combined test
simply means that both ANOVA and IMD are used, and results reported for
both tests.</p>
<p>Below, descriptions of ANOVA and IMD are provided, followed by a
description of the output from the <code><a href="../reference/imd_anova.html">imd_anova()</a></code>
function.</p>
</div>
<div class="section level2">
<h2 id="analysis-of-variance">Analysis of Variance<a class="anchor" aria-label="anchor" href="#analysis-of-variance"></a>
</h2>
<p>Quantitative tests for differential abundance of pan-omic data are
accomplished using the function <code>test_method = "anova"</code> or
<code>test_method = "combined"</code> arguments, which implement an
analysis of variance (ANOVA) for each biomolecule. <a href="https://CRAN.R-project.org/package=Rcpp" class="external-link"><code>Rcpp</code></a> and
parallel processing are used to speed up computation (<span class="citation">Eddelbuettel (2013)</span>). Different ANOVA
implementations are applied depending on the complexity of the supplied
data. For example, if only two groups are supplied then ANOVA reduces to
a two sample
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>t</mi><annotation encoding="application/x-tex">t</annotation></semantics></math>-test
that assumes equal variance for the two groups. Alternatively, if two
groups are present and a covariate correction is required, then a
two-factor ANCOVA is used to detect differences between all combinations
of groups or between main effects as appropriate controlling for
covariates. Details on generalizations to these two approaches are given
in the next two subsections.</p>
<div class="section level3">
<h3 id="one-group">One Group<a class="anchor" aria-label="anchor" href="#one-group"></a>
</h3>
<p>If there is only one grouping factor and that factor has more than
two levels, then a one-way ANOVA tests for quantitative differences
between the groups. Let
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>y</mi><mrow><mi>i</mi><mi>k</mi></mrow></msub><annotation encoding="application/x-tex">y_{ik}</annotation></semantics></math>
represent the biomolecule abundance for observation
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>k</mi><mo>=</mo><mn>1</mn><mo>,</mo><mi>…</mi><mo>,</mo><msub><mi>n</mi><mi>i</mi></msub></mrow><annotation encoding="application/x-tex">k=1,\dots,n_i</annotation></semantics></math>
in group
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>i</mi><mo>=</mo><mn>1</mn><mo>,</mo><mi>…</mi><mo>,</mo><mi>m</mi></mrow><annotation encoding="application/x-tex">i=1,\dots,m</annotation></semantics></math>
on the log base 2 scale then we assume
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>y</mi><mrow><mi>i</mi><mi>k</mi></mrow></msub><mo>=</mo><msub><mi>μ</mi><mi>i</mi></msub><mo>+</mo><msub><mi>ϵ</mi><mrow><mi>i</mi><mi>k</mi></mrow></msub></mrow><annotation encoding="application/x-tex">\begin{equation}
y_{ik}=\mu_i+\epsilon_{ik}
\label{eq:one_group}
\end{equation}</annotation></semantics></math> where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>μ</mi><mi>i</mi></msub><annotation encoding="application/x-tex">\mu_i</annotation></semantics></math>
represents the mean abundance of group
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>i</mi><annotation encoding="application/x-tex">i</annotation></semantics></math>
(on the log-scale) and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>ϵ</mi><mrow><mi>i</mi><mi>k</mi></mrow></msub><annotation encoding="application/x-tex">\epsilon_{ik}</annotation></semantics></math>
are the error terms that are assumed to be independent and identically
distributed (iid) according to the normal distribution with mean
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0</mn><annotation encoding="application/x-tex">0</annotation></semantics></math>
and variance
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>σ</mi><mn>2</mn></msup><annotation encoding="application/x-tex">\sigma^2</annotation></semantics></math>.
Group comparisons are performed according to <em>3.1 General linear
models</em> in (<span class="citation">Bretz, Hothorn, and Westfall
(2016)</span>). Under the hood, a linear model of the form
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>𝐲</mi><mo>=</mo><mi>𝐗</mi><mi>𝛃</mi><mo>+</mo><mi>ϵ</mi></mrow><annotation encoding="application/x-tex">\mathbf{y} = \mathbf{X}\mathbf{\beta} + \epsilon</annotation></semantics></math>
is fit, and the design matrix is used to form the appropriate contrast
matrix for comparing means. This framework extends to the two factor
case as well.</p>
<p>The code below will group the peptide data by Phenotype, which is a
categorical variable with three levels: “Phenotype1”, “Phenotype2”, and
“Phenotype3”. Then the <code>anova_test</code> function fits the model
described above and tests for differential abundance between each pair
of phenotypes for every peptide.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pep_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/group_designation.html">group_designation</a></span><span class="op">(</span></span>
<span>  omicsData <span class="op">=</span> <span class="va">pep_object</span>,</span>
<span>  main_effects <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Phenotype"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">myfilt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/imdanova_filter.html">imdanova_filter</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">pep_object</span><span class="op">)</span></span>
<span><span class="va">pep_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/applyFilt.html">applyFilt</a></span><span class="op">(</span></span>
<span>  filter_object <span class="op">=</span> <span class="va">myfilt</span>,</span>
<span>  omicsData <span class="op">=</span> <span class="va">pep_object</span>,</span>
<span>  min_nonmiss_anova <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  min_nonmiss_gtest <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span>
<span><span class="va">all_pairwise_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/imd_anova.html">imd_anova</a></span><span class="op">(</span></span>
<span>  omicsData <span class="op">=</span> <span class="va">pep_object</span>,</span>
<span>  test_method <span class="op">=</span> <span class="st">"anova"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The function <code>imd_anova</code> function returns a data frame
containing:</p>
<ol style="list-style-type: decimal">
<li><p>The number of observed values in each group in the columns called
<code>Count_i</code> where <code>i</code> is replaced by the group name
supplied as an attribute of the <code>omicsData</code>
argument,</p></li>
<li><p>The least squares means for each group in the columns called
<code>Mean_i</code> where <code>i</code> is replaced by the group name
supplied as an attribute of the <code>omicsData</code>
argument,</p></li>
<li><p>The
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>p</mi><annotation encoding="application/x-tex">p</annotation></semantics></math>-values
for each comparison in the columns called <code>P_value_A_</code> (“A”
for ANOVA)</p></li>
<li><p>The (log2) fold change for each comparison,</p></li>
<li><p>An indicator for significance for each comparison in the columns
called <code>Flag_A_</code> (“A” for ANOVA) that is a
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>±</mo><mn>1</mn></mrow><annotation encoding="application/x-tex">\pm1</annotation></semantics></math>
or 0 depending on whether that fold change was or was not statistically
significant from zero, respectively, according to the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>p</mi><annotation encoding="application/x-tex">p</annotation></semantics></math>-values
and the specified
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>p</mi><annotation encoding="application/x-tex">p</annotation></semantics></math>-value
threshold set by the <code>pval_thresh</code> argument to
<code>imd_anova</code>. A value of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>−</mo><mn>1</mn></mrow><annotation encoding="application/x-tex">-1</annotation></semantics></math>
implies the “control” or “reference” group is significantly under
expressed compared to the test group, and the converse is true for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>+</mo><mn>1</mn></mrow><annotation encoding="application/x-tex">+1</annotation></semantics></math>.</p></li>
</ol>
<p>The
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>p</mi><annotation encoding="application/x-tex">p</annotation></semantics></math>-values
associated with each group comparison can be adjusted for multiple
comparisons using the <code>pval_adjust_a_multcomp</code> argument to
<code>imd_anova</code>. A detailed look at the available
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>p</mi><annotation encoding="application/x-tex">p</annotation></semantics></math>-value
adjustment methods and when they should be applied are discussed later
in this vignette, but the use of the Tukey multiple comparison method is
illustrated in the following.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">all_pairwise_results_adjusted</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/imd_anova.html">imd_anova</a></span><span class="op">(</span></span>
<span>  omicsData <span class="op">=</span> <span class="va">pep_object</span>,</span>
<span>  test_method <span class="op">=</span> <span class="st">"anova"</span>,</span>
<span>  pval_adjust_a_multcomp <span class="op">=</span> <span class="st">"Tukey"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>By default, all pairwise comparisons are made. Custom comparisons can
by defined by passing a <code>data.frame</code> to the
<code>comparisons</code> argument with column names “Test” and
“Control”. Now, group “Phenotype1” is treated as the control group and
the remaining groups are compared to it.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">one_vs_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  Control <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Phenotype1"</span>, <span class="fl">2</span><span class="op">)</span>,</span>
<span>  Test <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Phenotype2"</span>, <span class="st">"Phenotype3"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">one_vs_all_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/imd_anova.html">imd_anova</a></span><span class="op">(</span></span>
<span>  omicsData <span class="op">=</span> <span class="va">pep_object</span>,</span>
<span>  test_method <span class="op">=</span> <span class="st">"anova"</span>,</span>
<span>  comparisons <span class="op">=</span> <span class="va">one_vs_all</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>If the grouping variable has only two levels,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>m</mi><mo>=</mo><mn>2</mn></mrow><annotation encoding="application/x-tex">m=2</annotation></semantics></math>,
then a Welch’s
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>t</mi><annotation encoding="application/x-tex">t</annotation></semantics></math>-test
is performed instead of an ANOVA. That is, the model defined in equation
() is still fit, but the two groups of the grouping factor are allowed
to have different variances. In particular, the error terms
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>ϵ</mi><mrow><mi>i</mi><mi>k</mi></mrow></msub><annotation encoding="application/x-tex">\epsilon_{ik}</annotation></semantics></math>
are assumed to be independently distributed according to the normal
distribution with mean
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0</mn><annotation encoding="application/x-tex">0</annotation></semantics></math>
and variance
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msubsup><mi>σ</mi><mi>i</mi><mn>2</mn></msubsup><annotation encoding="application/x-tex">\sigma^2_i</annotation></semantics></math>.
To determine if the biomolecules are differentially expressed,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>p</mi><annotation encoding="application/x-tex">p</annotation></semantics></math>-values
are derived from a
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>t</mi><annotation encoding="application/x-tex">t</annotation></semantics></math>-distribution
with degrees of freedom approximated by the Satterthwaite equation. The
value for <code>Variance</code>, which is not a pooled variance
estimate, scaled by the Satterthwaite approximate degrees of freedom. To
illustrate this the data are grouped by “SecondPhenotype”, which has two
values, then fit with ANOVA.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pep_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/group_designation.html">group_designation</a></span><span class="op">(</span></span>
<span>  omicsData <span class="op">=</span> <span class="va">pep_object</span>,</span>
<span>  main_effects <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SecondPhenotype"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">all_pairwise_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/imd_anova.html">imd_anova</a></span><span class="op">(</span></span>
<span>  omicsData <span class="op">=</span> <span class="va">pep_object</span>,</span>
<span>  test_method <span class="op">=</span> <span class="st">"anova"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>When this model is fit, it is imperative the researcher check the
assumptions of the model. In particular for the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>m</mi><mo>&gt;</mo><mn>2</mn></mrow><annotation encoding="application/x-tex">m&gt;2</annotation></semantics></math>
case, the residuals
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mover><mi>ϵ</mi><mo accent="true">̂</mo></mover><mrow><mi>i</mi><mi>k</mi></mrow></msub><mo>=</mo><msub><mi>y</mi><mrow><mi>i</mi><mi>k</mi></mrow></msub><mo>−</mo><msub><mover><mi>μ</mi><mo accent="true">̂</mo></mover><mi>i</mi></msub></mrow><annotation encoding="application/x-tex">\hat \epsilon_{ik}=y_{ik}-\hat \mu_i</annotation></semantics></math>
are assumed to be independent, normally distributed and have a common
variance across groups. The appropriateness of this assumption can be
assessed using quantile-quantile plots and comparisons of each group’s
variance.</p>
</div>
<div class="section level3">
<h3 id="two-groups">Two Groups<a class="anchor" aria-label="anchor" href="#two-groups"></a>
</h3>
<p>Next we consider the case where there are two grouping factors again
in the context of biomolecule abundance. Let
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>y</mi><mrow><mi>i</mi><mi>j</mi><mi>k</mi></mrow></msub><annotation encoding="application/x-tex">y_{ijk}</annotation></semantics></math>
represent the biomolecule abundance for observation
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>k</mi><mo>=</mo><mn>1</mn><mo>,</mo><mi>…</mi><mo>,</mo><msub><mi>n</mi><mrow><mi>i</mi><mi>j</mi></mrow></msub></mrow><annotation encoding="application/x-tex">k=1,\dots,n_{ij}</annotation></semantics></math>
in group
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>i</mi><mo>=</mo><mn>1</mn><mo>,</mo><mi>…</mi><mo>,</mo><mi>m</mi></mrow><annotation encoding="application/x-tex">i=1,\dots,m</annotation></semantics></math>
and group
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>j</mi><mo>=</mo><mn>1</mn><mo>,</mo><mi>…</mi><mo>,</mo><mi>p</mi></mrow><annotation encoding="application/x-tex">j=1,\dots,p</annotation></semantics></math>
on the log base 2 scale then we assume
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>y</mi><mrow><mi>i</mi><mi>j</mi><mi>k</mi></mrow></msub><mo>=</mo><msub><mi>μ</mi><mi>i</mi></msub><mo>+</mo><msub><mi>α</mi><mi>j</mi></msub><mo>+</mo><msub><mi>γ</mi><mrow><mi>i</mi><mi>j</mi></mrow></msub><mo>+</mo><msub><mi>ϵ</mi><mrow><mi>i</mi><mi>j</mi><mi>k</mi></mrow></msub></mrow><annotation encoding="application/x-tex">\begin{equation}
y_{ijk}=\mu_i+\alpha_j+\gamma_{ij}+\epsilon_{ijk}
\label{eq:full_model}
\end{equation}</annotation></semantics></math> where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>μ</mi><mi>i</mi></msub><mo>+</mo><msub><mi>α</mi><mi>j</mi></msub><mo>+</mo><msub><mi>γ</mi><mrow><mi>i</mi><mi>j</mi></mrow></msub></mrow><annotation encoding="application/x-tex">\mu_i+\alpha_j+\gamma_{ij}</annotation></semantics></math>
represents the mean abundance for an observation in
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>i</mi><annotation encoding="application/x-tex">i</annotation></semantics></math>th
first group and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>j</mi><annotation encoding="application/x-tex">j</annotation></semantics></math>th
second group which have marginal means
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>μ</mi><mi>i</mi></msub><annotation encoding="application/x-tex">\mu_i</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>α</mi><mi>j</mi></msub><annotation encoding="application/x-tex">\alpha_j</annotation></semantics></math>,
respectively. Finally,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>γ</mi><mrow><mi>i</mi><mi>j</mi></mrow></msub><annotation encoding="application/x-tex">\gamma_{ij}</annotation></semantics></math>
is the interaction effect of group
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>i</mi><mi>j</mi></mrow><annotation encoding="application/x-tex">ij</annotation></semantics></math>,
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>ϵ</mi><mrow><mi>i</mi><mi>j</mi><mi>k</mi></mrow></msub><annotation encoding="application/x-tex">\epsilon_{ijk}</annotation></semantics></math>
are the error terms that are assumed to be iid according to the normal
distribution with mean
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0</mn><annotation encoding="application/x-tex">0</annotation></semantics></math>
and variance
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>σ</mi><mn>2</mn></msup><annotation encoding="application/x-tex">\sigma^2</annotation></semantics></math>.</p>
<p>For each biomolecule, the <code>test_method = "anova"</code> argument
automatically tests if the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>γ</mi><mrow><mi>i</mi><mi>j</mi></mrow></msub><annotation encoding="application/x-tex">\gamma_{ij}</annotation></semantics></math>
effect is statistically significant from zero using a full and reduced
model
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>F</mi><annotation encoding="application/x-tex">F</annotation></semantics></math>-test.
That is, in addition to the model above, the reduced model
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>y</mi><mrow><mi>i</mi><mi>j</mi><mi>k</mi></mrow></msub><mo>=</mo><msub><mi>μ</mi><mi>i</mi></msub><mo>+</mo><msub><mi>α</mi><mi>j</mi></msub><mo>+</mo><msub><mi>ϵ</mi><mrow><mi>i</mi><mi>j</mi><mi>k</mi></mrow></msub></mrow><annotation encoding="application/x-tex">\begin{equation}
y_{ijk}=\mu_i+\alpha_j+\epsilon_{ijk}
\label{eq:red_model}
\end{equation}</annotation></semantics></math> is also fit and the test
statistic
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msup><mi>F</mi><mo>*</mo></msup><mo>=</mo><mfrac><mrow><mi>M</mi><mi>S</mi><msub><mi>E</mi><mi>F</mi></msub></mrow><mrow><mrow><mo stretchy="true" form="prefix">(</mo><mi>S</mi><mi>S</mi><msub><mi>E</mi><mi>R</mi></msub><mo>−</mo><mi>S</mi><mi>S</mi><msub><mi>E</mi><mi>F</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mi>/</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>d</mi><msub><mi>f</mi><mi>R</mi></msub><mo>−</mo><mi>d</mi><msub><mi>f</mi><mi>F</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow></mfrac></mrow><annotation encoding="application/x-tex">F^*=\frac{MSE_F}{(SSE_R-SSE_F)/(df_R-df_F)}</annotation></semantics></math>
is computed where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>M</mi><mi>S</mi><msub><mi>E</mi><mo>⋅</mo></msub></mrow><annotation encoding="application/x-tex">MSE_\cdot</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>d</mi><msub><mi>f</mi><mo>⋅</mo></msub></mrow><annotation encoding="application/x-tex">df_\cdot</annotation></semantics></math>
are the mean square error and degrees of freedom for the full (F; ) or
reduced (R; ) model, respectively. A
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>p</mi><annotation encoding="application/x-tex">p</annotation></semantics></math>-value
is computed based on
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>F</mi><mo>*</mo></msup><annotation encoding="application/x-tex">F^*</annotation></semantics></math>
as
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mo>=</mo><mi>P</mi><mrow><mo stretchy="true" form="prefix">(</mo><msup><mi>F</mi><mo>*</mo></msup><mo>&gt;</mo><msub><mi>F</mi><mrow><mi>a</mi><mo>,</mo><mi>b</mi></mrow></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p=P(F^*&gt;F_{a,b})</annotation></semantics></math>
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>a</mi><mo>,</mo><mi>b</mi></mrow></msub><annotation encoding="application/x-tex">F_{a,b}</annotation></semantics></math>
is compared to an
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>F</mi><annotation encoding="application/x-tex">F</annotation></semantics></math>
distribution with degrees of freedom
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>a</mi><mo>=</mo><mi>d</mi><msub><mi>f</mi><mi>F</mi></msub></mrow><annotation encoding="application/x-tex">a=df_F</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>b</mi><mo>=</mo><mi>d</mi><msub><mi>f</mi><mi>R</mi></msub><mo>−</mo><mi>d</mi><msub><mi>f</mi><mi>F</mi></msub></mrow><annotation encoding="application/x-tex">b=df_R-df_F</annotation></semantics></math>.
If
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mo>&lt;</mo><mn>0.05</mn></mrow><annotation encoding="application/x-tex">p&lt;0.05</annotation></semantics></math>
then there is significant evidence to reject the null hypothesis that
the reduced model is adequate and the full model is used to test for
group differences. Conversely, if
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mo>≥</mo><mn>0.05</mn></mrow><annotation encoding="application/x-tex">p\geq 0.05</annotation></semantics></math>
then there is not enough evidence to reject the null hypothesis and the
reduced model is used. Group comparisons are computed in the same manner
as the one-factor case, following <em>3.1 General linear models</em> in
(<span class="citation">Bretz, Hothorn, and Westfall (2016)</span>).</p>
<p>Regardless of whether the full or reduced model is used, both assume
that the residuals are independent and normally distributed with a
common variance for all groups. These assumptions must be checked before
the results of the model are used to assess differential abundance.</p>
</div>
<div class="section level3">
<h3 id="covariate-adjustment">Covariate Adjustment<a class="anchor" aria-label="anchor" href="#covariate-adjustment"></a>
</h3>
<p>If, in addition to a single grouping factor, there are covariates to
account for when testing for differential abundance then a linear model
of the form
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>𝐲</mi><mo>=</mo><mi>𝐗</mi><msub><mi>β</mi><mn>1</mn></msub><mo>+</mo><mi>𝐕</mi><msub><mi>β</mi><mn>2</mn></msub><mo>+</mo><mi>ϵ</mi></mrow><annotation encoding="application/x-tex">\begin{equation}
\mathbf{y}=\mathbf{X} \beta_1+\mathbf{V} \beta_2 + \epsilon
\label{eq:covar_adj}
\end{equation}</annotation></semantics></math> is fit to the transformed
abundance data
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐲</mi><annotation encoding="application/x-tex">\mathbf{y}</annotation></semantics></math>
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐗</mi><annotation encoding="application/x-tex">\mathbf{X}</annotation></semantics></math>
is the dummy-encoded design matrix for the main effects,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐕</mi><annotation encoding="application/x-tex">\mathbf{V}</annotation></semantics></math>
is the design matrix for any covariates,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>β</mi><mn>1</mn></msub><annotation encoding="application/x-tex">\beta_1</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>β</mi><mn>2</mn></msub><annotation encoding="application/x-tex">\beta_2</annotation></semantics></math>
are the parameters to be estimated for the main effects and covariates,
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ϵ</mi><annotation encoding="application/x-tex">\epsilon</annotation></semantics></math>
is the error term. Note that
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐗</mi><annotation encoding="application/x-tex">\mathbf{X}</annotation></semantics></math>
allows for the one factor model, as well as the full and reduced forms
of the two factor model. We estimate the least squares means using the
estimated coefficients
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>β</mi><mn>1</mn></msub><annotation encoding="application/x-tex">\beta_1</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>β</mi><mn>2</mn></msub><annotation encoding="application/x-tex">\beta_2</annotation></semantics></math>.
Specifically, when computing the the least squares mean for a particular
combination of main effect levels, we consider the predicted value of
the fitted model at the main effect levels in question, the mean value
of continuous covariates, and averaged over all levels of categorical
covariates. This is equivalent to the treatment in the
<code>emmeans</code> package (<span class="citation">Lenth
(2023)</span>).</p>
</div>
<div class="section level3">
<h3 id="paired-data">Paired Data<a class="anchor" aria-label="anchor" href="#paired-data"></a>
</h3>
<p>The comparison of samples at two points in time from the same
individual (e.g., a before and after state) is common in omics
experiments and referred to as “paired data.” To enable analysis of
paired data in <code>pmartR</code>, we have implemented a method for
specifying pair information and added appropriate code to any
<code>pmartR</code> functions impacted by the use of paired data.</p>
<p>If data are paired, this should be specified using the
<code>group_designation</code> function. Users can also designate which
member of a pair is the “control” or “reference.”</p>
<p>ANOVA tests can be performed on paired data with or without any main
effects or covariates.</p>
<p>The robust Mahalanobis distance (rMd) filter for sample outlier
detection has been updated to remove all parts of a pair when at least
one of the individuals in the pair does not pass the specified p-value
threshold. Filters for biomolecules have also been updated to consider
the pair as opposed to each individual member of the pair.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># add fake pair info to f_data using SecondPhenotype</span></span>
<span><span class="va">pep_object</span><span class="op">$</span><span class="va">f_data</span><span class="op">$</span><span class="va">Pairing</span> <span class="op">&lt;-</span> <span class="va">pep_object</span><span class="op">$</span><span class="va">f_data</span><span class="op">$</span><span class="va">SecondPhenotype</span></span>
<span><span class="va">pep_object</span><span class="op">$</span><span class="va">f_data</span><span class="op">$</span><span class="va">Pair_ID</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, <span class="fl">2</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">5</span><span class="op">:</span><span class="fl">8</span>, <span class="fl">2</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">9</span><span class="op">:</span><span class="fl">12</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mypaired</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/group_designation.html">group_designation</a></span><span class="op">(</span></span>
<span>  omicsData <span class="op">=</span> <span class="va">pep_object</span>,</span>
<span>  pair_group <span class="op">=</span> <span class="st">"Pairing"</span>,</span>
<span>  pair_id <span class="op">=</span> <span class="st">"Pair_ID"</span>,</span>
<span>  pair_denom <span class="op">=</span> <span class="st">"B"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">myfilt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/imdanova_filter.html">imdanova_filter</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mypaired</span><span class="op">)</span></span>
<span><span class="va">mypaired</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/applyFilt.html">applyFilt</a></span><span class="op">(</span></span>
<span>  filter_object <span class="op">=</span> <span class="va">myfilt</span>,</span>
<span>  omicsData <span class="op">=</span> <span class="va">mypaired</span>,</span>
<span>  min_nonmiss_anova <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  min_nonmiss_gtest <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">paired_stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/imd_anova.html">imd_anova</a></span><span class="op">(</span></span>
<span>  omicsData <span class="op">=</span> <span class="va">mypaired</span>,</span>
<span>  test_method <span class="op">=</span> <span class="st">"anova"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="test-for-independence-of-missing-data">Test for Independence of Missing Data<a class="anchor" aria-label="anchor" href="#test-for-independence-of-missing-data"></a>
</h2>
<p>In the event that there aren’t enough observed values to test for a
quantitative difference in abundance between groups, one could still
test for a <em>qualitative</em> difference in groups using the
independence of missing data (IMD) test. This is often the case for
proteomics data where peptides or proteins could have missing data for
one of several reasons. The idea is to assess if there are more missing
data in one group compared to another. If there are an adequate number
of non-missing data available, then the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>c</mi><mi>h</mi><msup><mi>i</mi><mn>2</mn></msup></mrow><annotation encoding="application/x-tex">chi^2</annotation></semantics></math>
test of independence can be used. This assumption often fails, however,
so a modified version of the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>c</mi><mi>h</mi><msup><mi>i</mi><mn>2</mn></msup></mrow><annotation encoding="application/x-tex">chi^2</annotation></semantics></math>
test, called the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>G</mi><annotation encoding="application/x-tex">G</annotation></semantics></math>-test,
should be used.</p>
<p>The test statistic associated with the IMD test for each biomolecule
is given by
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>G</mi><mo>=</mo><mn>2</mn><munderover><mo>∑</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></munderover><mrow><mo stretchy="true" form="prefix">[</mo><msub><mi>C</mi><mrow><mi>O</mi><mi>k</mi></mrow></msub><mo>ln</mo><mrow><mo stretchy="true" form="prefix">(</mo><mfrac><msub><mi>C</mi><mrow><mi>O</mi><mi>k</mi></mrow></msub><msub><mi>E</mi><mrow><mi>O</mi><mi>k</mi></mrow></msub></mfrac><mo stretchy="true" form="postfix">)</mo></mrow><mo>+</mo><msub><mi>C</mi><mrow><mi>A</mi><mi>k</mi></mrow></msub><mo>ln</mo><mrow><mo stretchy="true" form="prefix">(</mo><mfrac><msub><mi>C</mi><mrow><mi>A</mi><mi>k</mi></mrow></msub><msub><mi>E</mi><mrow><mi>A</mi><mi>k</mi></mrow></msub></mfrac><mo stretchy="true" form="postfix">)</mo></mrow><mo stretchy="true" form="postfix">]</mo></mrow></mrow><annotation encoding="application/x-tex">\begin{equation}
G=2\sum_{k=1}^K\left[C_{Ok}\ln\left(\frac{C_{Ok}}{E_{Ok}}\right)+C_{Ak}\ln\left(\frac{C_{Ak}}{E_{Ak}}\right) \right]
\label{eq:gtest}
\end{equation}</annotation></semantics></math> where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>C</mi><mrow><mi>O</mi><mi>k</mi></mrow></msub><annotation encoding="application/x-tex">C_{Ok}</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>E</mi><mrow><mi>O</mi><mi>k</mi></mrow></msub><annotation encoding="application/x-tex">E_{Ok}</annotation></semantics></math>
are the observed and expected number of non-missing values for group
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>,
respectively. The quantities
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>C</mi><mrow><mi>A</mi><mi>k</mi></mrow></msub><annotation encoding="application/x-tex">C_{Ak}</annotation></semantics></math><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>E</mi><mrow><mi>A</mi><mi>k</mi></mrow></msub><annotation encoding="application/x-tex">E_{Ak}</annotation></semantics></math>
are similarly defined for the missing (absent) values. Further, the
expected number of non-missing values is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>E</mi><mrow><mi>O</mi><mi>k</mi></mrow></msub><mo>=</mo><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>m</mi><mi>O</mi></msub><msub><mi>n</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mi>/</mi><mi>N</mi></mrow><annotation encoding="application/x-tex">E_{Ok}=(m_On_k)/N</annotation></semantics></math>
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>n</mi><mi>k</mi></msub><annotation encoding="application/x-tex">n_k</annotation></semantics></math>
is the number of samples associated with group
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>N</mi><annotation encoding="application/x-tex">N</annotation></semantics></math>
is the total number of samples and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>m</mi><mi>O</mi></msub><annotation encoding="application/x-tex">m_O</annotation></semantics></math>
is the number of samples with this biomolecule missing.</p>
<p>The function <code>test_method = "gtest"</code> or
<code>test_method = "combined"</code> argument specification will
perform the test for missing data. The results object returned is the
same as described above, for ANOVA, except an “_G” is used to denote
columns specific to the g-test.</p>
<ol style="list-style-type: decimal">
<li><p>The number of observed values in each group in the columns called
<code>Count_i</code> where <code>i</code> is replaced by the group name
supplied as an attribute of the <code>omicsData</code>
argument,</p></li>
<li><p>The best linear unbiased estimators of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>μ</mi><mi>i</mi></msub><annotation encoding="application/x-tex">\mu_i</annotation></semantics></math>
in the columns called <code>Mean_i</code> where <code>i</code> is
replaced by the group name supplied as an attribute of the
<code>omicsData</code> argument,</p></li>
<li><p>The
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>p</mi><annotation encoding="application/x-tex">p</annotation></semantics></math>-values
for each comparison in the columns called <code>P_value_G_</code> (“G”
g-test)</p></li>
<li><p>The (log2) fold change for each comparison,</p></li>
<li><p>An indicator for significance for each comparison in the columns
called <code>Flag_G_</code> (“G” g-test) that is a
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>±</mo><mn>1</mn></mrow><annotation encoding="application/x-tex">\pm1</annotation></semantics></math>
or 0 depending on whether the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>p</mi><annotation encoding="application/x-tex">p</annotation></semantics></math>-values
were significant or not. In this case, a value of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>−</mo><mn>1</mn></mrow><annotation encoding="application/x-tex">-1</annotation></semantics></math>
implies the “control” or “reference” group had fewer observations than
the test group, and the converse is true for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>+</mo><mn>1</mn></mrow><annotation encoding="application/x-tex">+1</annotation></semantics></math>.</p></li>
</ol>
<p>The
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>p</mi><annotation encoding="application/x-tex">p</annotation></semantics></math>-values
associated with each group comparison can be adjusted for multiple
comparisons using the <code>pval_adjust_g_multcomp</code> argument to
<code>imd_anova</code>.</p>
<div class="section level3">
<h3 id="covariate-adjustment-1">Covariate Adjustment<a class="anchor" aria-label="anchor" href="#covariate-adjustment-1"></a>
</h3>
<p>Covariates can also be accounted for in the g-test. This is
accomplished internally by passing their terms to the model
specification in <code>glm</code>, which is used to compute the IMD
test.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mypep_covariates</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/group_designation.html">group_designation</a></span><span class="op">(</span><span class="va">pep_object</span>, main_effects <span class="op">=</span> <span class="st">"Phenotype"</span>, covariates <span class="op">=</span> <span class="st">"Characteristic"</span><span class="op">)</span></span>
<span><span class="va">imd_covariates</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/imd_anova.html">imd_anova</a></span><span class="op">(</span></span>
<span>  omicsData <span class="op">=</span> <span class="va">mypep_covariates</span>,</span>
<span>  test_method <span class="op">=</span> <span class="st">"gtest"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="paired-data-1">Paired Data<a class="anchor" aria-label="anchor" href="#paired-data-1"></a>
</h3>
<p>The g-test in the imd-ANOVA test can now be run on paired data and
requires a “main effects” designation, unlike the quantitative test
(ANOVA).</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># add fake pair info to f_data using SecondPhenotype</span></span>
<span><span class="va">mypaired</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/group_designation.html">group_designation</a></span><span class="op">(</span></span>
<span>  omicsData <span class="op">=</span> <span class="va">pep_object</span>,</span>
<span>  main_effects <span class="op">=</span> <span class="st">"Phenotype"</span>,</span>
<span>  pair_group <span class="op">=</span> <span class="st">"Pairing"</span>,</span>
<span>  pair_id <span class="op">=</span> <span class="st">"Pair_ID"</span>,</span>
<span>  pair_denom <span class="op">=</span> <span class="st">"B"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">myfilt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/imdanova_filter.html">imdanova_filter</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mypaired</span><span class="op">)</span></span>
<span><span class="va">mypaired</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/applyFilt.html">applyFilt</a></span><span class="op">(</span></span>
<span>  filter_object <span class="op">=</span> <span class="va">myfilt</span>,</span>
<span>  omicsData <span class="op">=</span> <span class="va">mypaired</span>,</span>
<span>  min_nonmiss_anova <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  min_nonmiss_gtest <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">paired_stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/imd_anova.html">imd_anova</a></span><span class="op">(</span></span>
<span>  omicsData <span class="op">=</span> <span class="va">mypaired</span>,</span>
<span>  test_method <span class="op">=</span> <span class="st">"gtest"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="output">Output<a class="anchor" aria-label="anchor" href="#output"></a>
</h2>
<p>The <code><a href="../reference/imd_anova.html">imd_anova()</a></code> function returns an object of class
<code>statRes</code> that is a data frame with some attributes. The data
frame includes:</p>
<ul>
<li>Biomolecule name</li>
<li>Counts of observed values per main effect group</li>
<li>Least squares means of observed values per main effect group</li>
<li>log2 fold changes for each comparison</li>
<li>p-values for each comparison and test</li>
<li>Flags indicating significance</li>
</ul>
<p>The <code>statRes</code> object also consists of several attributes
which specify the methods used to arrive at the results. This way, the
results object is self-contained and can be passed to other functions
for further analysis with ease. In particular, the attributes are</p>
<ul>
<li>
<code>group_DF</code> - <code>data.frame</code> defining which
samples are associated with which groups</li>
<li>
<code>comparisons</code> - <code>data.frame</code> defining the
comparisons that were tested</li>
<li>
<code>number_significant</code> - <code>data.frame</code> giving the
number of significant biomolecules (up and down) for each
comparison</li>
<li>
<code>statistical_test</code> - character string giving the
statistical test(s) run</li>
<li>
<code>adjustment_method_a</code> and
<code>adjustment_method_g</code> - character string giving the
adjustment for multiple comparisons that was used; NA if none was
used</li>
<li>
<code>pval_thresh</code> - the numeric value that was an input to
<code>anova_test</code>
</li>
<li>
<code>which_X</code> - For two factor models, a vector of zeros and
ones indicating whether the corresponding biomolecule had the full model
with interaction terms fit (ones) or the reduced model without
interaction terms fit (zeros). For one factor models, this is a vector
of zeros.</li>
</ul>
<p>Several methods are defined for objects of class <code>statRes</code>
including <code>summary</code>, <code>print</code> and
<code>plot</code>, which are discussed in more depth in the next
section.</p>
</div>
<div class="section level2">
<h2 id="methods-for-results-objects">Methods for Results Objects<a class="anchor" aria-label="anchor" href="#methods-for-results-objects"></a>
</h2>
<p>Special functions are available for objects of class
<code>statRes</code> including <code>print</code>, <code>summary</code>
and <code>plot</code>. Due to the typical size the <code>statRes</code>
objects, only the head of each element in a <code>statRes</code> object
is printed along with the summarized attributes; use
<code>print.default</code> to see the full object. The
<code>summary</code> functions prints the type of test that was run, any
adjustments that were made, the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>p</mi><annotation encoding="application/x-tex">p</annotation></semantics></math>-value
threshold used to define significance and a table that summarizes the
number of significant biomolecules (up or down) for each comparison. See
below for an illustration.</p>
<p>The <code>plot</code> function can be used to produce any subset of
the following four plots specified by the <code>plot_type</code>
argument. If not <code>plot_type</code> is supplied, then all four are
created, otherwise on the subset specified is produced.</p>
<ul>
<li>
<code>"bar"</code> - shows the number of significant biomolecules
grouped by comparisons</li>
<li>
<code>"volcano"</code> - a plot of the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>p</mi><annotation encoding="application/x-tex">p</annotation></semantics></math>-value
by the fold-change estimate differentiated by test and faceted by
comparison</li>
<li>
<code>"heatmap"</code> - illustrates the fold changes for the
statistically significant biomolecules (only available if more than one
comparison is done)</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stat_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/imd_anova.html">imd_anova</a></span><span class="op">(</span></span>
<span>  omicsData <span class="op">=</span> <span class="va">pep_object</span>,</span>
<span>  test_method <span class="op">=</span> <span class="st">"combined"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">stat_results</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Type of test: combined </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Multiple comparison adjustment ANOVA: </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Multiple comparison adjustment G-test: </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## p-value threshold: 0.05 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Number of significant biomolecules by comparison. Columns specify fold change direction and type of test:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##   Comparison Total:Positive Total:Negative Positive:ANOVA Negative:ANOVA</span></span>
<span><span class="co">## 1     A_vs_B           2062            287           1038            108</span></span>
<span><span class="co">##   Positive:G-test Negative:G-test</span></span>
<span><span class="co">## 1            1024             179</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">stat_results</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## statRes</span></span>
<span><span class="co">##                         Peptide Count_A Count_B   Mean_A   Mean_B</span></span>
<span><span class="co">## 1              AAAAAAAAAAGAAGGR       4       6 26.82222 26.19385</span></span>
<span><span class="co">## 2                   AAAAAAALQAK       3       3 25.76461 26.51143</span></span>
<span><span class="co">## 3     AAAAAEQQQFYLLLGNLLSPDNVVR       8       8 25.02951 24.83093</span></span>
<span><span class="co">## 4                  AAAAAGTATSQR      10      11 25.35116 25.37603</span></span>
<span><span class="co">## 5                          ----    ----    ----     ----     ----</span></span>
<span><span class="co">## 18103            YTIIIPENLKPQMK       5       1 26.16265 26.18524</span></span>
<span><span class="co">## 18104                  YTQELTLK       3       2 25.07080 24.42717</span></span>
<span><span class="co">## 18105             YTYSEWHSFTQPR       2       2 25.21807 25.38213</span></span>
<span><span class="co">## 18106            YVYDSAFHPDTGEK       3       2 25.07202 24.09574</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">stat_results</span>, stacked<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="Statistical_Analysis_with_pmartR_files/figure-html/statres-1.png" width="576"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">stat_results</span>, plot_type <span class="op">=</span> <span class="st">"volcano"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Statistical_Analysis_with_pmartR_files/figure-html/statres-2.png" width="576"></p>
</div>
<div class="section level2">
<h2 id="p-value-adjustment-methods">
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>p</mi><annotation encoding="application/x-tex">p</annotation></semantics></math>-value
Adjustment Methods<a class="anchor" aria-label="anchor" href="#p-value-adjustment-methods"></a>
</h2>
<p>If several comparisons are being made it is a good idea to adjust the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>p</mi><annotation encoding="application/x-tex">p</annotation></semantics></math>-values
in order to control the false positive rate. Currently
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>p</mi><annotation encoding="application/x-tex">p</annotation></semantics></math>-value
adjustments are made at the biomolecule level (i.e., adjustments are
made for the number of groups being compared). The available
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>p</mi><annotation encoding="application/x-tex">p</annotation></semantics></math>-value
adjustments are described in the following table. The “Appropriate
Comparison” column refers to which correction method is preferred based
on the type of comparisons being made, i.e, all pairwise or
case-to-control. Check marks in the “ANOVA” and “IMD” columns indicate
if that correction method is appropriate for ANOVA or IMD
comparisons.</p>
<table class="table">
<thead><tr class="header">
<th>Method Name</th>
<th>Appropriate Comparison</th>
<th>ANOVA</th>
<th>IMD</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Bonferroni</td>
<td>Both</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msqrt><mrow></mrow></msqrt><annotation encoding="application/x-tex">\sqrt{}</annotation></semantics></math></td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msqrt><mrow></mrow></msqrt><annotation encoding="application/x-tex">\sqrt{}</annotation></semantics></math></td>
</tr>
<tr class="even">
<td>Dunnett</td>
<td>Case-vs-control</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msqrt><mrow></mrow></msqrt><annotation encoding="application/x-tex">\sqrt{}</annotation></semantics></math></td>
<td></td>
</tr>
<tr class="odd">
<td>Holm</td>
<td>Both</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msqrt><mrow></mrow></msqrt><annotation encoding="application/x-tex">\sqrt{}</annotation></semantics></math></td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msqrt><mrow></mrow></msqrt><annotation encoding="application/x-tex">\sqrt{}</annotation></semantics></math></td>
</tr>
<tr class="even">
<td>Tukey</td>
<td>All pairwise</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msqrt><mrow></mrow></msqrt><annotation encoding="application/x-tex">\sqrt{}</annotation></semantics></math></td>
<td></td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="false-discovery-rate-methods">False Discovery Rate Methods<a class="anchor" aria-label="anchor" href="#false-discovery-rate-methods"></a>
</h2>
<p>To perform false discovery rate (FDR) control, we can pass the
argument <code>pval_adjust_a_fdr</code> or
<code>pval_adjust_g_fdr</code> for the anova and IMD tests respectively.
Options are “bonferroni”, “BH”, “BY”, and “fdr”, see
<code><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">?p.adjust</a></code>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stat_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/imd_anova.html">imd_anova</a></span><span class="op">(</span></span>
<span>  omicsData <span class="op">=</span> <span class="va">pep_object</span>,</span>
<span>  test_method <span class="op">=</span> <span class="st">"combined"</span>,</span>
<span>  pval_adjust_a_fdr <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>  pval_adjust_g_fdr <span class="op">=</span> <span class="st">"BH"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2 unnumbered">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-bretz_multcomp" class="csl-entry">
Bretz, Frank, Torsten Hothorn, and Peter Westfall. 2016. <em>Multiple
Comparisons Using r</em>. CRC Press.
</div>
<div id="ref-rcpp_book" class="csl-entry">
Eddelbuettel, Dirk. 2013. <em>Seamless <span>R</span> and
<span>C++</span> Integration with <span>Rcpp</span></em>. New York:
Springer.
</div>
<div id="ref-emmeans" class="csl-entry">
Lenth, Russell V. 2023. <em>Emmeans: Estimated Marginal Means, Aka
Least-Squares Means</em>. <a href="https://CRAN.R-project.org/package=emmeans" class="external-link">https://CRAN.R-project.org/package=emmeans</a>.
</div>
<div id="ref-webb2010combined" class="csl-entry">
Webb-Robertson, Bobbie-Jo M, Lee Ann McCue, Katrina M Waters, Melissa M
Matzke, Jon M Jacobs, Thomas O Metz, Susan M Varnum, and Joel G Pounds.
2010. <span>“Combined Statistical Analyses of Peptide Intensities and
Peptide Occurrences Improves Identification of Significant Peptides from
MS-Based Proteomics Data.”</span> <em>Journal of Proteome Research</em>
9 (11): 5748–56.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Lisa Bramer, Kelly Stratton, Daniel Claborne.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
